tests/TestCase/Datasource/Log/ElasticLoggerTest.php

```diff
@@ -0,0 +1,484 @@
+use Cake\Database\Log\QueryLogger;
+use Cake\ElasticSearch\Datasource\Connection;
+use Cake\ElasticSearch\Datasource\Log\ElasticLogger;
+use Cake\ElasticSearch\TestSuite\TestCase;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+
+/**
+ * Tests the ElasticLogger class with simple array format
+ */
+class ElasticLoggerTest extends TestCase
+{
+    /**
+     * Test ElasticLogger constructor
+     */
+    public function testConstruct(): void
+    {
+        $queryLogger = new QueryLogger();
+        $connection = $this->getMockConnection();
+
+        $elasticLogger = new ElasticLogger($queryLogger, $connection);
+
+        $this->assertSame($queryLogger, $elasticLogger->getLogger());
+        $this->assertInstanceOf(ElasticLogger::class, $elasticLogger);
+    }
+
+    /**
+     * Test constructor with PSR-3 LoggerInterface
+     */
+    public function testConstructWithPsrLogger(): void
+    {
+        $psrLogger = $this->createMock(LoggerInterface::class);
+        $connection = $this->getMockConnection();
+
+        $elasticLogger = new ElasticLogger($psrLogger, $connection);
+
+        $this->assertSame($psrLogger, $elasticLogger->getLogger());
+    }
+
+    /**
+     * Test setLogger and getLogger methods
+     */
+    public function testSetAndGetLogger(): void
+    {
+        $initialLogger = new QueryLogger();
+        $connection = $this->getMockConnection();
+        $elasticLogger = new ElasticLogger($initialLogger, $connection);
+
+        // Test initial logger
+        $this->assertSame($initialLogger, $elasticLogger->getLogger());
+
+        // Test setting new logger
+        $newLogger = $this->createMock(LoggerInterface::class);
+        $result = $elasticLogger->setLogger($newLogger);
+
+        $this->assertSame($elasticLogger, $result); // Test fluent interface
+        $this->assertSame($newLogger, $elasticLogger->getLogger());
+    }
+
+    /**
+     * Test log method when query logging is enabled with DEBUG level
+     */
+    public function testLogWhenLoggingEnabledDebugLevel(): void
+    {
+        $mockLogger = $this->createMock(LoggerInterface::class);
+        $connection = $this->getMockConnection(true);
+        $elasticLogger = new ElasticLogger($mockLogger, $connection);
+
+        $context = [
+            'request' => [
+                'size' => 50,
+                'from' => 0,
+                'query' => ['match_all' => []],
+            ],
+        ];
+
+        $mockLogger->expects($this->once())
+            ->method('log')
+            ->with(
+                LogLevel::DEBUG,
+                $this->callback(function ($message) {
+                    $data = json_decode($message, true);
+
+                    return isset($data['size']) && $data['size'] === 50 &&
+                           isset($data['from']) && $data['from'] === 0 &&
+                           isset($data['query']['match_all']);
+                }),
+                $this->anything(),
+            );
```

Comment:
Sometimes it is simpler to use `ArrayLog` to capture all the log messages and then you can do assertions on the order and content.